docs/guides/table_migration.md

```diff
@@ -0,0 +1,120 @@
+4. Create a new primary [`INCREMENTAL_BY_TIME_RANGE` model](../concepts/models/model_kinds.md#incremental_by_time_range) named `my_schema.existing_table` (see below for code).
+5. Run `sqlmesh plan` to create and backfill the models
+
+The staging model would contain code similar to:
+
+``` sql linenums="1"
+MODEL(
+    name my_schema.existing_table_staging,
+    kind INCREMENTAL_BY_TIME_RANGE(
+        time_column table_time_column
+    )
+)
+
+SELECT
+    col1,
+    col2,
+    col3
+FROM
+    [your model's ongoing data source]
+WHERE
+    table_time_column BETWEEN @start_ds and @end_ds;
+```
+
+The primary model would contain code similar to:
+
+``` sql linenums="1"
+MODEL(
+    name my_schema.existing_table,
+    kind INCREMENTAL_BY_TIME_RANGE(
+        time_column table_time_column
+    )
+)
+
+SELECT
+    col1,
+    col2,
+    col3
+FROM
+    my_schema.existing_table_staging
+WHERE
+    table_time_column BETWEEN @start_ds and @end_ds
+UNION
+SELECT
+    col1,
+    col2,
+    col3
+FROM
+    my_schema.existing_table_historical
+WHERE
+    table_time_column BETWEEN @start_ds and @end_ds;
+```
+
+When `sqlmesh plan` is first run on these models, the entire historical table will be read into the primary model. Future executions with `sqlmesh run` will return no rows because the historical table will not contain new data after the model is first created.
+
+If the model is modified, `sqlmesh plan` will again read the entire historical table into the model so it can properly apply the updated model code.
+
+
+## Snapshot replacement
+
+The snapshot replacement method works by renaming an existing table to a name that SQLMesh recognizes as part of an existing SQLMesh model.
+
+We begin this section by briefly describing how SQLMesh's virtual data environments work. Conceptually, SQLMesh divides the database into a "physical layer" where data is stored and a "virtual layer" where data is accessed by end users. The physical layer stores materialized objects like tables, and the virtual layer contains views that point to the physical layer objects.
+
+Each time a SQLMesh `plan` adds or modifies a model, SQLMesh creates a physical layer "snapshot" object to which the virtual layer view points. The snapshot replacement method simply renames the existing table to migrate to the name of the appropriate snapshot table.
+
+For example, consider an existing table named `my_schema.existing_table`. Migrating this table with the snapshot replacement method involves five steps:
+
+1. Ensure `my_schema.existing_table` is up to date (has ingested all available source data)
+2. Rename `my_schema.existing_table` to any other name, such as `my_schema.existing_table_temp`.
+3. Create and initialize an empty [`INCREMENTAL_BY_TIME_RANGE` model](../concepts/models/model_kinds.md#incremental_by_time_range) named `my_schema.existing_table`:
+- Make the model [forward only](./incremental_time.md#forward-only-models) by setting the `MODEL` DDL key `forward_only` to `true`
```

Comment:
Worth explaining why this is needed - in order to not override the historical data.